boehmgc: 8.0.6 -> 8.2.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/metapackage/boehm-gc/versions.


meta.description for boehmgc is: "The Boehm-Demers-Weiser conservative garbage collector for C and C++"

meta.homepage for boehmgc is: "https://hboehm.info/gc/"

meta.changelog for boehmgc is: ""


###### Updates performed
- Version update

###### To inspect upstream changes

- [Release on GitHub](https://github.com/ivmai/bdwgc/releases/tag/v8.2.0)

- [Compare changes on GitHub](https://github.com/ivmai/bdwgc/compare/v8.0.6...v8.2.0)

###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 8.2.0 with grep in /nix/store/7dw1b46p4jv05h6nr02693w14j8p5m8d-boehm-gc-8.2.0
- found 8.2.0 in filename of file in /nix/store/7dw1b46p4jv05h6nr02693w14j8p5m8d-boehm-gc-8.2.0

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
18871 total rebuild path(s)

13061 package rebuild(s)

13043 x86_64-linux rebuild(s)
5828 x86_64-darwin rebuild(s)


First fifty rebuilds by attrpath
ArchiSteamFarm
AusweisApp2
CHOWTapeModel
ChowKick
CuboCore.coreaction
CuboCore.corearchiver
CuboCore.corefm
CuboCore.coregarage
CuboCore.corehunt
CuboCore.coreimage
CuboCore.coreinfo
CuboCore.corekeyboard
CuboCore.corepad
CuboCore.corepaint
CuboCore.corepdf
CuboCore.corepins
CuboCore.corerenamer
CuboCore.coreshot
CuboCore.corestats
CuboCore.corestuff
CuboCore.coreterminal
CuboCore.coretime
CuboCore.coretoppings
CuboCore.coreuniverse
CuboCore.libcprime
CuboCore.libcsys
DisnixWebService
EmptyEpsilon
MIDIVisualizer
MMA
OSCAR
OVMF
OVMFFull
R
SDL
SDL2
SDL2_gfx
SDL2_image
SDL2_mixer
SDL2_net
SDL2_ttf
SDL_Pango
SDL_gfx
SDL_gpu
SDL_image
SDL_mixer
SDL_net
SDL_sixel
SDL_sound
SDL_stretch
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/7dw1b46p4jv05h6nr02693w14j8p5m8d-boehm-gc-8.2.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A boehmgc https://github.com/r-ryantm/nixpkgs/archive/d971b042a31d989cce62b8a6805ee5703daa491b.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/7dw1b46p4jv05h6nr02693w14j8p5m8d-boehm-gc-8.2.0
ls -la /nix/store/7dw1b46p4jv05h6nr02693w14j8p5m8d-boehm-gc-8.2.0/bin
```

---

</details>
<br/>



### Pre-merge build results

NixPkgs review skipped

---

###### Maintainer pings

r-ryantm committed:
boehmgc: 8.0.6 -> 8.2.0

mohe2015 commented:
Take care when testing as nix depends on this afaik.

mohe2015 commented on pkgs/development/libraries/boehm-gc/default.nix:
```diff
@@ -1,18 +1,18 @@
 { lib, stdenv, fetchurl
 , autoreconfHook
-, enableLargeConfig ? false # doc: https://github.com/ivmai/bdwgc/blob/v8.0.6/doc/README.macros#L195
+, enableLargeConfig ? false # doc: https://github.com/ivmai/bdwgc/blob/v8.2.0/doc/README.macros#L195
```
```suggestion
, enableLargeConfig ? false # doc: https://github.com/ivmai/bdwgc/blob/v8.2.0/doc/README.macros#L199
```

roberth committed:
boehmgc: Make comment future proof

It gets updated to new tags automatically but the line number would
have to change.

roberth committed:
boehmgc: Add Nix packages to passthru

NickCao commented:
Look's like this is a pre-release, and it is causing troubles, ref: https://github.com/ivmai/bdwgc/issues/388, fixed but did not make it's way into release.

bobby285271 commented:
https://github.com/ivmai/bdwgc/commit/47e9106c17b72e9ee5501308f69ea94531e798b3 explicitly says this is an experimental release, I guess we should revert this on staging-next. Has been reported to repology.

trofi commented:
8.2.0 also misses a header in includes (9f3f822984a3ec939de6dd0f4f632e486d451ad6) and breaks guile-3 builds:

```
error: builder for '/nix/store/v3vkr9vaxg278mdi6cv2a0jaqw8mn3xp-guile-3.0.7.drv' failed with exit code 2;
       last 10 log lines:
       >                  from alist.c:31:
       > /nix/store/2dzi3ls093ffivpp6ykhb3wh33ngyaww-boehm-gc-8.2.0-dev/include/gc/gc.h:1784:11: fatal error: gc_pthread_redirects.h: No such file or directory
       >  1784 | # include "gc_pthread_redirects.h"
       >       |           ^~~~~~~~~~~~~~~~~~~~~~~~
```

I suggest a revert for now.

trofi commented:
Proposed a revert as https://github.com/NixOS/nixpkgs/pull/152205
